include/meta/class-bulk-delete-post-meta.php

```diff
@@ -297,16 +312,30 @@ public static function delete_post_meta( $delete_options ) {
 			);
 		}
 
-		if ( $use_value ) {
-			$options['meta_query'] = apply_filters( 'bd_delete_post_meta_query', array(), $delete_options );
-		} else {
+		if ( 'use_key' === $use_value ) {
 			$options['meta_key'] = $meta_key;
+		} else {
+			$options['meta_query'] = apply_filters( 'bd_delete_post_meta_query', array(), $delete_options );
 		}
 
 		$post_ids = bd_query( $options );
 		foreach ( $post_ids as $post_id ) {
-			if ( delete_post_meta( $post_id, $meta_key ) ) {
-				$count++;
+
+			if ( isset( $delete_options['meta_keys'] ) && is_array( $delete_options['meta_keys'] ) ) {
+				$is_post_id_counted_flag = false;
```

Comment:
@mariadanieldeepak 

You can remove `_flag` at end. It doesn't add any value to the name.